src/hooks/useFeatureFlags.ts

```diff
@@ -1,15 +1,38 @@
 import { FFContext } from '../context/FFContext'
-import { useContext } from 'react'
-import { FlagIdentifiers, getFlagsSubset } from '../util/getFlagsSubset'
+import { useContext, useMemo } from 'react'
 
 export const useFeatureFlags = (
-  flagIdentifiers?: FlagIdentifiers
+  flagIdentifiers?: Record<string, any> | string[]
 ): Record<keyof typeof flagIdentifiers, any> => {
-  const { flags } = useContext(FFContext)
+  const { client, flags } = useContext(FFContext)
 
-  if (flagIdentifiers) {
-    return getFlagsSubset(flags, flagIdentifiers)
-  }
+  return useMemo(() => {
+    if (!client) {
+      return {}
+    }
 
-  return flags
+    if (flagIdentifiers) {
+      if (Array.isArray(flagIdentifiers)) {
```

Comment:
could this be shortened to ` if (flagIdentifiers && Array.isArray(flagIdentifiers))`, also NBD

Reply:
Not really... the outer `if` surrounds this and the `else`. I'd need to change the `else` to an `else if (flagIdentifiers)`.

I could flip the logic and return the default return down below if `!flagIdentifiers`

Reply:
I found another issue, so will be in the neighbourhood anyway, so might as well... I'm not a fan of `if`s in `if`s if it can be helped

Reply:
yeah ofc I totally...spotted...that....